stubs/inertia/resources/js/Jetstream/Dropdown.vue

```diff
@@ -55,13 +55,19 @@
                 }
             }
 
-            this.$once('hook:destroyed', () => {
-                document.removeEventListener('keydown', closeOnEscape)
-            })
-
             document.addEventListener('keydown', closeOnEscape)
         },
 
+        unmounted() {
+            const closeOnEscape = (e) => {
```

Comment:
Hmm, does redefining the entire function actually work when removing it? I don't think so. I think you need the reference to the original function to remove it.

Here's what I believe is the proper Vue 3 way to handle this. Remove both the `mounted()` and `unmounted()` methods, and then add a new `setup()` method instead. Something like this:

```js
import { onMounted, onUnmounted } from 'vue'

setup () {
  const closeOnEscape = (e) => {
      if (this.open && e.keyCode === 27) {
          this.open = false
      }
  }
  
  onMounted(() => document.addEventListener('keydown', closeOnEscape))
  onUnmounted(() => document.removeEventListener('keydown', closeOnEscape))
},
```

Again, this is untested, so definitely do test it, but this should work. 👍